### ISSUE TITLE
Exclude animation dock widget from viewer screenshot

### ISSUE THREAD

**Issue by sofroniewn** [sofroniewn/napari-animation#23]
One thing that might be nice is to figure out how to exclude the animation dock widget from the viewer screenshot if it is not wanted. I guess the widget can always be undocked, but it is something we might want to consider

**Comment by alisterburt** [comment]
I guess the use case is for creating animations showing how to do things in the viewer/the effect of changing things in the viewer? Makes sense, will have a think

**Comment by alisterburt** [comment]
@katherine-hutchings @cormsby626 this would be a slightly more technical place to contribute which could be fun!

The whole dock widget will need to be hidden, not just the `AnimationWidget` which lives inside the dock widget -  something like
`viewer.window._dock_widgets['animation: Wizard'].hide()`
`viewer.window._dock_widgets['animation: Wizard'].show()`
will get us going but it would be good to see if we can achieve this using the public API... maybe @tlambert03 or @DragaDoncila knows if there is a nice public API for hiding dock widgets?

As before, feel free to ping me any time if you choose to take this on, either here or on the [napari zulip](https://napari.zulipchat.com)

**Comment by tlambert03** [comment]
> maybe @tlambert03 or @DragaDoncila knows if there is a nice public API for hiding dock widgets?

nope, but I'd say it's ultimately the same request as https://github.com/napari/napari/issues/2203 and https://github.com/napari/napari/issues/3944

**Comment by katherine-hutchings** [comment]
Hi Guys I had a go at implementing this in the way suggested, however it did not seem to work as expected.

Hiding the instance of `QtViewerDockWidget` from the integrated ipython console works as expected, but when this is called during the animate method the canvas does not resize itself to take up the extra space.
<img width="600" alt="image" src="https://user-images.githubusercontent.com/112178739/187464237-9acbc03a-3df7-455d-9935-180047aaec8b.png">

I tried to make a minimal example in napari, but could not reproduce so the issue must be something to do with napari-animation.

```python
import napari
from magicgui import magicgui

viewer = napari.Viewer()

widget_name = 'widget'

@magicgui
def my_func():
    viewer.window._dock_widgets[widget_name].hide()


viewer.window.add_dock_widget(my_func, name=widget_name)
napari.run()
```

@tlambert03 @czaki - @alisterburt mentioned you guys know a lot about Qt, is this something you immediately know the cause of?

**Comment by tlambert03** [comment]
yeah, hiding a specific dock widget will do just that: hide the dock widget.  But what you really need to manipulate here is the QMainWindow instance itself...  specifically, with [removeDockWidget](https://doc.qt.io/qt-6/qmainwindow.html#removeDockWidget) and `restoreDockWidget`.  You could do this with a context manager

```python
from contextlib import contextmanager

@contextmanager
def hidden_dock_widget(window: QMainWindow, widget: QDockWidget):
    window.removeDockWidget(widget)
    try:
        yield
    finally:
        window.restoreDockWidget(widget)
```

```python
with hidden_dock_widget(viewer.window._qt_window, your_dock_widget):
    # render
```

**Comment by katherine-hutchings** [comment]
Thank you very much, I will look into this :)
